core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -400,7 +398,9 @@ private[kafka] class Processor(val id: Int,
 
   private val newConnections = new ConcurrentLinkedQueue[SocketChannel]()
   private val inflightResponses = mutable.Map[String, RequestChannel.Response]()
-  private val metricTags = Map("networkProcessor" -> id.toString).asJava
+  val metricTags = Map(
+    "protocol" -> securityProtocol.name, "listener" -> listenerName.value, "networkProcessor" -> id.toString
+    ).asJava
```

Comment:
Yes, let's just remove the security protocol algother. Synced with Jun and he agrees that this is fine.